File: United_Community/Amit/Daily-Leetcode/day_14_median_of_two_sorted_arrays.cpp
//Try-1
class Solution {
public:
    double findMedianSortedArrays(vector<int>& nums1, vector<int>& nums2) {
        vector<int> res;
        double ans;
        int i = 0, j = 0, k = 0;
        while (i != nums1.size() && j!= nums2.size()) {
            if (nums1[i] < nums2[j]) {
                res.push_back(nums1[i]);
                i++;
            }
            else if (nums1[i] > nums2[j]) {
                res.push_back(nums2[j]);
                j++;
            }
            else {
                res.push_back(nums1[i]);
                i++;
                res.push_back(nums2[j]);
                j++;
            }
        }
        while (i < nums1.size()) {
            res.push_back(nums1[i]);
            i++;
        }
        while (j < nums2.size()) {
            res.push_back(nums2[j]);
            j++;
        }
        //cout << (res[res.size() / 2 - 1] + res[res.size() / 2]);
        if (res.size() % 2 == 0) {
            ans = (double)(res[res.size() / 2 - 1] + res[res.size() / 2]) / 2.0;
        }
        else {
            ans = (double)res[res.size() / 2];
        }
        return ans;
    }
};

//Try-2 - Even more time
// class Solution {
// public:
//     double findMedianSortedArrays(vector<int>& nums1, vector<int>& nums2) {
//         vector<int> res;
//         double ans;
//         for(int i = 0; i < nums1.size(); i++) {
//             res.push_back(nums1[i]);
//         }
//         for (int i = 0; i < nums2.size(); i++) {
//             res.push_back(nums2[i]);
//         }
//         sort(res.begin(), res.end());
//         if (res.size() % 2 == 0) {
//             ans = (double)(res[res.size() / 2 - 1] + res[res.size() / 2]) / 2.0;
//         }
//         else {
//             ans = (double)res[res.size() / 2];
//         }
//         return ans;
//     }
// }; 

File: United_Community/Mradul/Daily-Leetcode/day_14_MedianOfTwoSortedArrays.java
// https://leetcode.com/problems/median-of-two-sorted-arrays/

class Solution {
    public double findMedianSortedArrays(int[] nums1, int[] nums2) {
        int n = nums1.length, m = nums2.length, k =0;
        int[] arr = new int[n + m];

        for (int i = 0; i < n; i++) {
            arr[i] = nums1[i];
        }
        for (int i = 0; i < m; i++) {
            arr[i + n] = nums2[i];
        }
        Arrays.sort(arr);
        int size = arr.length;

        if (size % 2 != 0) {
            return (double)arr[size/2];
        }
        else {
            int m1 = arr[size/2];
            int m2 = arr[size/2 - 1];
            return (double)(m1 + m2)/2.0;
        }
    }
}